src/wasm/wasm-validator.cpp

```diff
@@ -3477,61 +3479,79 @@ void FunctionValidator::visitStringSliceWTF(StringSliceWTF* curr) {
                "string operations require reference-types [--enable-strings]");
 }
 
+void FunctionValidator::visitContNew(ContNew* curr) {
+  // TODO implement actual type-checking
+  shouldBeTrue(!getModule() || getModule()->features.hasStackSwitching(),
+               curr,
+               "cont.new requires stack-switching [--enable-stack-switching]");
+
+  shouldBeTrue((curr->type.isContinuation() &&
+                curr->type.getHeapType().getContinuation().type.isSignature()),
+               curr,
+               "invalid type in ContNew expression");
+}
+
 void FunctionValidator::visitContBind(ContBind* curr) {
   // TODO implement actual type-checking
-  shouldBeTrue(
-    !getModule() || getModule()->features.hasTypedContinuations(),
-    curr,
-    "cont.bind requires typed-continuatons [--enable-typed-continuations]");
+  shouldBeTrue(!getModule() || getModule()->features.hasStackSwitching(),
+               curr,
+               "cont.bind requires stack-switching [--enable-stack-switching]");
 
-  shouldBeTrue((curr->contTypeBefore.isContinuation() &&
-                curr->contTypeBefore.getContinuation().type.isSignature()),
+  shouldBeTrue((curr->sourceType.isContinuation() &&
+                curr->sourceType.getContinuation().type.isSignature()),
                curr,
                "invalid first type in ContBind expression");
 
-  shouldBeTrue((curr->contTypeAfter.isContinuation() &&
-                curr->contTypeAfter.getContinuation().type.isSignature()),
+  shouldBeTrue((curr->type.isContinuation() &&
+                curr->type.getHeapType().getContinuation().type.isSignature()),
                curr,
                "invalid second type in ContBind expression");
 }
 
-void FunctionValidator::visitContNew(ContNew* curr) {
+void FunctionValidator::visitSuspend(Suspend* curr) {
```

Comment:
It would still be good to validate that `sendTypes` is the right size. We'll just have to require the sent types to be calculated when the expression is first built.